File: frontend/src/app/common/entity/purchase.ts
import {Customer} from "./customer";
import {Address} from "./address";
import {Order} from "./order";
import {OrderItem} from "./orderItem";

export class Purchase {
  customer: Customer;
  shippingAddress: Address;
  billingAddress: Address;
  order: Order;
  orderItems: OrderItem[];


  constructor(customer: Customer, shippingAddress: Address, billingAddress: Address, order: Order, orderItems: OrderItem[]) {
    this.customer = customer;
    this.shippingAddress = shippingAddress;
    this.billingAddress = billingAddress;
    this.order = order;
    this.orderItems = orderItems;
  }
}

File: backend/src/main/java/com/pazdev/guitarshop/dto/Purchase.java
package com.pazdev.guitarshop.dto;

import com.pazdev.guitarshop.entity.Address;
import com.pazdev.guitarshop.entity.Customer;
import com.pazdev.guitarshop.entity.Order;
import com.pazdev.guitarshop.entity.OrderItem;
import lombok.Data;

import java.util.Set;

@Data
public class Purchase {
    private Customer customer;
    private Address shippingAddress;
    private Address billingAddress;
    private Order order;
    private Set<OrderItem> orderItems;
}
